executor/table_reader.go

```diff
@@ -210,6 +212,13 @@ func (e *TableReaderExecutor) buildResp(ctx context.Context, ranges []*ranger.Ra
 	if err != nil {
 		return nil, err
 	}
+	if len(e.dagPB.Executors) == 2 {
```

Comment:
This only affects TableReader. How about involving this upon IndexScan,  IndexLookup and IndexMerge?

Reply:
The `Concurrency` should also be set to 1 if Limit is not pushed down.

@qw4990 Do you have any suggestions to implement this?